packages/components/src/collection/src/getElementRef.tsx

```diff
@@ -0,0 +1,10 @@
+import { ReactElement, Ref } from "react";
+
+/**
+ * `ref` is passed as prop in React 19, whereas `ref` is directly attached to
+ * children in React 18 below check is to ensure `ref` is accessible in both
+ * cases
+ */
+export function getElementRef(element: ReactElement): Ref<any> {
```

Comment:
Function copied from https://github.com/mui/material-ui/pull/42818/files#diff-d984286486e9f102e06bf15b0838b605dad169ef2770665abaa0eb7264b574db

This works in React 19, and shouldn't throw a warning in React 18.